models/src/main/scala/coop/rchain/casper/PrettyPrinter.scala

```diff
@@ -10,23 +10,31 @@ object PrettyPrinter {
   def buildStringNoLimit(b: Array[Byte]): String = Base16.encode(b)
   def buildStringNoLimit(b: ByteString): String  = Base16.encode(b.toByteArray)
 
-  def buildString(t: CasperMessage): String =
+  def buildString(t: CasperMessage, short: Boolean = false): String =
     t match {
-      case b: BlockMessage => buildString(b)
+      case b: BlockMessage => buildString(b, short)
       case _               => "Unknown consensus protocol message"
     }
 
   // TODO shouldn header.parentsHashList be nonempty list?
-  private def buildString(b: BlockMessage): String =
-    b.header.parentsHashList.headOption
-      .fold(s"Block ${buildString(b.blockHash)} with missing elements")(
-        mainParent =>
-          s"Block #${b.body.state.blockNumber} (${buildString(b.blockHash)}) " +
-            s"-- Sender ID ${buildString(b.sender)} " +
-            s"-- M Parent Hash ${buildString(mainParent)} " +
-            s"-- Contents ${buildString(b.body.state)}" +
-            s"-- Shard ID ${limit(b.shardId, 10)}"
-      )
+  private def buildString(b: BlockMessage, short: Boolean): String =
+    short match {
+      case false =>
+        b.header.parentsHashList.headOption
+          .fold(s"Block ${buildString(b.blockHash)} with missing elements")(
+            mainParent =>
+              s"Block #${b.body.state.blockNumber} (${buildString(b.blockHash)}) " +
+                s"-- Sender ID ${buildString(b.sender)} " +
+                s"-- M Parent Hash ${buildString(mainParent)} " +
+                s"-- Contents ${buildString(b.body.state)}" +
+                s"-- Shard ID ${limit(b.shardId, 10)}"
+          )
+      case true =>
+        b.header.parentsHashList.headOption
+          .fold(s"Block ${buildString(b.blockHash)} with missing elements")(
+            mainParent => s"#${b.body.state.blockNumber} (${buildString(b.blockHash)})"
+          )
+    }
```

Comment:
Why not just simple if..else? And also inside of fold because empty parent case is the same.